src/libstd/io/stdio.rs

```diff
@@ -872,6 +874,93 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that a prompt
+/// is printed before capturing input. This function returns
+/// either success ([`Ok`]) or failure ([`Err`]) containing
+/// the input.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// # Panics
+///
+/// This function currently panics if it does not receive input,
+/// or when input is empty (end of file / `ctrl + d`).
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(prompt)]
+/// use std::io::prompt;
+///
+/// fn main() {
+///     let name = prompt("Enter name: ").expect("input failed!");
+///
+///     println!("Your name is {}!", name);
+/// }
+/// ```
+#[unstable(
+    feature = "prompt",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn prompt(prompt: &str) -> Result<String> {
+    stdout().write(prompt.as_bytes())?;
+    input()
+}
+
+/// Constructs a new handle to the standard input of the current
+/// process, locks this handle and reads a line of input, returning
+/// either success ([`Ok`]) or failure ([`Err`]) containing the input.
+/// For automatic prompt handling, see the [`prompt`] method.
+///
+/// If you need more explicit control over
+/// locking, see the [`Stdin::lock`] and [`Stdout::lock`] methods.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
```

Comment:
Same as above, I don't think we need to so strongly say it's not recommended.

Reply:
trimmed notes